cmd/gcp-controller-manager/kubelet_readonly_csr_approver_test.go

```diff
@@ -384,6 +386,56 @@ func TestValidateRbac(t *testing.T) {
 	}
 }
 
+func TestValidateCommonName(t *testing.T) {
+	testCases := []struct {
+		name         string
+		commonName   string
+		expectResult kubeletReadonlyCSRResponse
+	}{
+		{
+			name:       fmt.Sprintf("x509cr has prefix %s", kubeletReadonlyCRCommonName),
+			commonName: kubeletReadonlyCRCommonName,
+			expectResult: kubeletReadonlyCSRResponse{
+				result:  true,
+				err:     nil,
+				message: fmt.Sprintf("x509 common name start with %s", kubeletReadonlyCRCommonName),
+			},
+		},
+		{
+			name:       fmt.Sprintf("x509cr not has prefix %s", kubeletReadonlyCRCommonName),
+			commonName: "test",
+			expectResult: kubeletReadonlyCSRResponse{
+				result:  false,
+				err:     nil,
+				message: fmt.Sprintf("x509 common name not start with %s", kubeletReadonlyCRCommonName),
+			},
+		},
+		{
+			name: "x509 is empty",
```

Comment:
I don't think you are testing the case wherr x509 result is nil or empty I believe in this case only the CommonName is empty

Reply:
Yes, line 460 if the common name is empty, x509 will not be set